internal/exec/stack_processor_utils.go

```diff
@@ -272,46 +328,23 @@ func ProcessYAMLConfigFileWithContext(
 	}
 
 	stackManifestTemplatesProcessed := stackYamlConfig
-	stackManifestTemplatesErrorMessage := ""
 
-	// Process `Go` templates in the imported stack manifest using the provided `context`
+	// Process `Go` templates in the imported stack manifest based on file type and context.
 	// https://atmos.tools/core-concepts/stacks/imports#go-templates-in-imports
-	if !skipTemplatesProcessingInImports && len(context) > 0 {
+	if ShouldProcessFileAsTemplate(filePath, context, skipTemplatesProcessingInImports) {
 		stackManifestTemplatesProcessed, err = ProcessTmpl(relativeFilePath, stackYamlConfig, context, ignoreMissingTemplateValues)
 		if err != nil {
-			if atmosConfig.Logs.Level == u.LogLevelTrace || atmosConfig.Logs.Level == u.LogLevelDebug {
-				stackManifestTemplatesErrorMessage = fmt.Sprintf("\n\n%s", stackYamlConfig)
-			}
-			// Check if we have merge context to provide enhanced error formatting
-			if mergeContext != nil {
-				// Wrap the error with the sentinel first to preserve it
-				wrappedErr := fmt.Errorf("%w: %v", errUtils.ErrInvalidStackManifest, err)
-				// Then format it with context information
-				e := mergeContext.FormatError(wrappedErr, fmt.Sprintf("stack manifest '%s'%s", relativeFilePath, stackManifestTemplatesErrorMessage))
-				return nil, nil, nil, nil, nil, nil, nil, e
-			} else {
-				e := fmt.Errorf("%w: stack manifest '%s'\n%v%s", errUtils.ErrInvalidStackManifest, relativeFilePath, err, stackManifestTemplatesErrorMessage)
-				return nil, nil, nil, nil, nil, nil, nil, e
-			}
+			return nil, nil, nil, nil, nil, nil, nil, formatTemplateProcessingError(
+				err, atmosConfig, mergeContext, relativeFilePath, stackYamlConfig,
+			)
 		}
 	}
 
 	stackConfigMap, err := u.UnmarshalYAMLFromFile[schema.AtmosSectionMapType](atmosConfig, stackManifestTemplatesProcessed, filePath)
 	if err != nil {
-		if atmosConfig.Logs.Level == u.LogLevelTrace || atmosConfig.Logs.Level == u.LogLevelDebug {
-			stackManifestTemplatesErrorMessage = fmt.Sprintf("\n\n%s", stackYamlConfig)
-		}
-		// Check if we have merge context to provide enhanced error formatting
-		if mergeContext != nil {
-			// Wrap the error with the sentinel first to preserve it
-			wrappedErr := fmt.Errorf("%w: %v", errUtils.ErrInvalidStackManifest, err)
-			// Then format it with context information
-			e := mergeContext.FormatError(wrappedErr, fmt.Sprintf("stack manifest '%s'%s", relativeFilePath, stackManifestTemplatesErrorMessage))
-			return nil, nil, nil, nil, nil, nil, nil, e
-		} else {
-			e := fmt.Errorf("%w: stack manifest '%s'\n%v%s", errUtils.ErrInvalidStackManifest, relativeFilePath, err, stackManifestTemplatesErrorMessage)
-			return nil, nil, nil, nil, nil, nil, nil, e
-		}
```

Comment:
Note to reviewer, this was moved to formatTemplateProcessingError so we can test it.